components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/step/impl/DefaultStepHandler.java

```diff
@@ -1421,4 +1431,36 @@ private void setLoggedInOrgIdInRequest(AuthenticatedIdPData authenticatedIdPData
             }
         }
     }
+
+    /**
+     * Resolve user resident organization for the users authenticated via the B2B organization SSO authenticator.
+     *
+     * @param authenticatedUser   The authenticated user.
+     * @param authenticatorConfig The authenticator configurations.
+     * @return The organization where the user's identity is managed.
+     */
+    private String resolveUserResidentOrganization(AuthenticatedUser authenticatedUser,
+                                                   AuthenticatorConfig authenticatorConfig) {
+
+        // Check for user organization claim for the authenticated user via the organization SSO authenticator.
+        if (authenticatedUser.getUserAttributes() != null && isLoggedInWithOrganizationSSO(authenticatorConfig)) {
+            for (Map.Entry<ClaimMapping, String> userAttributes : authenticatedUser.getUserAttributes().entrySet()) {
+                if (FrameworkConstants.USER_ORGANIZATION_CLAIM.equals(
+                        userAttributes.getKey().getLocalClaim().getClaimUri())) {
+                    return userAttributes.getValue();
+                }
+            }
+        }
+        return StringUtils.EMPTY;
```

Comment:
what is the scenario we get here?

Reply:
Fixed by - https://github.com/wso2/carbon-identity-framework/pull/4915/commits/7360e490007132ad9850fab163818ce8ba5a938c
This method is only executed for B2B apps. If B2C user logs from a B2B app, the organization ID of the tenant domain is returned.